core/state/statedb.go

```diff
@@ -78,7 +83,7 @@ type DB struct {
 	stateObjects        map[common.Address]*Object
 	stateObjectsPending map[common.Address]struct{} // State objects finalized but not yet written to the trie
 	stateObjectsDirty   map[common.Address]struct{}
-	stateValidators     map[common.Address]*stk.ValidatorWrapper
+	stateValidators     *lru.Cache
```

Comment:
This stateValidators is like a staged representation of the validatorWrappers in memory for easy modification without having to serialize/deserialize from the account.code field everything it's modified. This map keeps track of all the modifications which will needs to be committed to stateDB eventually. Changing it to cache with a limit could potentially lose modificaiton data for the validatorWrappers.

Reply:
Good point. I propose setting `validatorWrapperCacheLimit` to 4,000 (currently the PR has it set to 1,000) to work around this issue. The rationale for doing so is that the block gas limit is 80,000,000, and the minimum cost of a staking transaction can be ~20,000 (conservatively; it is often higher). This means that at most 4,000 validator wrapper modifications can occur in a block. Alternatively, I am happy to change it back to a dictionary format. Let me know what you think.

Reply:
Could you please change stateValidators to a data structure with something like

```
type validatorCache struct {
    dirty map[common.Address]*ValidatorWrapper
    cache *lru.Cache
}
```

This will make it more readable and will not add unnecessary number of caches to the data structure.

Reply:
Based on the new profiling, seems there isn't much benefit in using LRU. Blockchain system requires strong deterministic guarantee, please revert to the old way of using maps. Since each block state will be cleared after the block is processed, there shouldn't be memory leak issue on the old way.

Reply:
I saw you added the dirty/cache struct. Since it's not improving any memory performance. Let's not complicate the existing code (may introduce new bugs and it's risky without extensive testing). Sorry for letting you change the code back and forth. (And also it's too complicated to have three boolean in the ValidatorWrapper() method.)